x-pack/plugins/security/server/authentication/api_keys/api_keys.ts

```diff
@@ -335,6 +337,32 @@ export class APIKeys {
     return result;
   }
 
+  /**
+   * Tries to validate an API key.
+   * @param apiKey ApiKey.
+   */
+  async validate(apiKey: Pick<CreateAPIKeyResult, 'id' | 'api_key'>) {
+    if (!this.license.isEnabled()) {
+      return { isValid: false };
+    }
+
+    const fakeRequest = getFakeKibanaRequest(apiKey);
+
+    this.logger.debug(`Trying to validate an API key as current user`);
+
+    let result: SecurityAuthenticateResponse;
+
+    try {
+      result = await this.clusterClient.asScoped(fakeRequest).asCurrentUser.security.authenticate();
+      this.logger.debug(`API key was validated successfully`);
+      return { isValid: true, result };
+    } catch (e) {
+      this.logger.info(`Failed to validate API key ${e.message}`);
```

Comment:
optional super-nit for consistency with the rest of this service:

```suggestion
      this.logger.info(`Failed to validate API key: ${e.message}`);
```